src/lib/alloc.c

```diff
@@ -99,17 +117,54 @@ static inline uint32_t heap_get_size(struct mm_heap *heap)
 	return size;
 }
 
-#if DEBUG_BLOCK_ALLOC || DEBUG_BLOCK_FREE
-static void alloc_memset_region(void *ptr, uint32_t bytes, uint32_t val)
+#if DEBUG_BLOCK_FREE
+static void write_pattern(struct mm_heap *heap_map, int heap_depth,
+						  uint8_t pattern)
 {
-	uint32_t count = bytes >> 2;
-	uint32_t *dest = ptr, i;
+	struct mm_heap *heap;
+	struct block_map *current_map;
+	int i, j;
 
-	for (i = 0; i < count; i++)
-		dest[i] = val;
+	for (i = 0; i < heap_depth; i++) {
+		heap = &heap_map[i];
+
+		for (j = 0; j < heap->blocks; j++) {
+			current_map = &heap->map[j];
+			memset(
+				(void *)current_map->base, DEBUG_BLOCK_FREE_VALUE,
```

Comment:
Thinking aloud, if DEBUG_BLOCK_FREE_VALUE was always guaranteed to be an invalid memory address (and hence ptr) for each platform then any deref would cause an exception (where the handler could invoke debugger/dump trace etc).

Reply:
cache will also need to be written back and invalidated.

Reply:
You mean in a situation where our pattern is dereferenced ? :D A nice thought - I will try to come up with something for that.

In this situation if we are using cashable address and do not operate on different cores. 
I don't  think we need to written back.